api/src/main/java/org/apache/iceberg/metrics/MultiDimensionCounterKey.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+// TODO gaborkaszab: comment
+public interface MultiDimensionCounterKey {
+  String get();
+
+  int numDimensions();
```

Comment:
I'm a bit confused why this method is needed

Reply:
I figured that we should guarantee that an instance of a MultiDimensionCounter has to work on a certain number of dimensions. I made the number of dimensions configurable through constructor and then whenever I give a MetricTag to the MultiDimensionCounter (e.g. when I call insert()) I validate that the MetricTag has the same number of dimensions as the MultiDimensionCounter.